From dbe386af0161939f7b32d592dd41d17398ee6610 Mon Sep 17 00:00:00 2001
From: Leith Bade <leith@bade.nz>
Date: Thu, 27 Jun 2024 17:58:53 +1000
Subject: [PATCH 1/4] arm64: dts: mediatek: mt7986: fix the switch reset line
 on BPI-R3

The current GPIO definition for the MT7531 switch reset line incorrectly
specifies a pin GPIO_0 (GPIO5) that is connected to the boot mode
selection DIP switch (SW1).

In the public schematic for the BPI-R3 (titled "BPI-R3-MT7986A",
revision "V1.1", sheet 4), the SPI1_CS (GPIO32) pin of the MT7986 (U1E)
is connected to the GbE_RESET net.

Frank Wunderlich has told me, via the Banana Pi forum, that this GbE_RESET
net connects to the MT7531 reset line in the private part of the shematic
he has a copy of.

Signed-off-by: Leith Bade <leith@bade.nz>
---
 arch/arm64/boot/dts/mediatek/mt7986a-bananapi-bpi-r3.dts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/mediatek/mt7986a-bananapi-bpi-r3.dts b/arch/arm64/boot/dts/mediatek/mt7986a-bananapi-bpi-r3.dts
index e7654dc9a1c9..cead58aaca59 100644
--- a/arch/arm64/boot/dts/mediatek/mt7986a-bananapi-bpi-r3.dts
+++ b/arch/arm64/boot/dts/mediatek/mt7986a-bananapi-bpi-r3.dts
@@ -203,7 +203,7 @@ switch: switch@31 {
 		interrupt-parent = <&pio>;
 		interrupts = <66 IRQ_TYPE_LEVEL_HIGH>;
 		#interrupt-cells = <1>;
-		reset-gpios = <&pio 5 GPIO_ACTIVE_HIGH>;
+		reset-gpios = <&pio 32 GPIO_ACTIVE_HIGH>;
 	};
 };
 
-- 
2.51.2

